File: php/features.php
<?php
declare(strict_types=1);

// SpacexRest SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class SpacexRestFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new SpacexRestBaseFeature();
            case "test":
                return new SpacexRestTestFeature();
            default:
                return new SpacexRestBaseFeature();
        }
    }
}

File: rb/features.rb
# SpacexRest SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module SpacexRestFeatures
  def self.make_feature(name)
    case name
    when "base"
      SpacexRestBaseFeature.new
    when "test"
      SpacexRestTestFeature.new
    else
      SpacexRestBaseFeature.new
    end
  end
end
